x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/use_actions.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { pick } from 'lodash';
+
+import { EuiTableActionsColumnType } from '@elastic/eui';
+
+import { CreateAnalyticsFormProps } from '../../hooks/use_create_analytics_form';
+import { CloneButton } from '../action_clone';
+import { useDeleteAction, DeleteButton, DeleteButtonModal } from '../action_delete';
+import { useStartAction, StartButton, StartButtonModal } from '../action_start';
+import { StopButton } from '../action_stop';
+import { getViewAction } from '../action_view';
+
+import { isDataFrameAnalyticsRunning, DataFrameAnalyticsListRow } from './common';
+
+export const useActions = (
+  createAnalyticsForm: CreateAnalyticsFormProps,
+  isManagementTable: boolean
+): {
+  actions: EuiTableActionsColumnType<DataFrameAnalyticsListRow>['actions'];
+  modals: JSX.Element | null;
+} => {
+  const deleteAction = useDeleteAction();
+  const startAction = useStartAction();
+
+  let modals: JSX.Element | null = null;
+
+  const actions: EuiTableActionsColumnType<DataFrameAnalyticsListRow>['actions'] = [
+    getViewAction(isManagementTable),
+  ];
+
+  if (isManagementTable === false) {
+    modals = (
+      <>
+        <StartButtonModal
+          {...pick(startAction, ['closeModal', 'isModalVisible', 'item', 'startAndCloseModal'])}
```

Comment:
nit: You can avoid using `lodash` with destructuring

Reply:
Destructuring would result in having to list all attributes twice wouldn't it? I decided to get rid of the fine grained picking and passing on just certain props and instead pass along the full object that we get back from the hook even if not all are used, makes the code much less verbose and simpler. Updated in 4382fee.